src/component/ResponsiveContainer.tsx

```diff
@@ -68,9 +68,25 @@ export const ResponsiveContainer = forwardRef<HTMLDivElement, Props>(
     const containerRef = useRef<HTMLDivElement>(null);
     const onResizeRef = useRef<Props['onResize']>();
     onResizeRef.current = onResize;
-    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-    // @ts-ignore
-    useImperativeHandle(ref, () => containerRef);
+    useImperativeHandle(ref, () => {
+      return Object.assign(containerRef.current, {
+        current: new Proxy(containerRef.current, {
```

Comment:
this returns a proxy object which will break users that currently have `ref.current.current` int their code

```jsx
    React.useEffect(() => {
      if (ref) {
        console.log(ref.current.current);
        console.log(ref.current.current.children);
      }
    }, [ref]);
```

This works in TypeScript, but it does not execute because current.current is a proxy object 

![image](https://github.com/recharts/recharts/assets/25180830/4e68c8b0-2f31-4548-9c58-89f0920c51d9)

Reply:
@ckifer Could you please try it once again???